### PR TITLE
added a "bin_dir" parameter to configure where the Kafka scripts are

### PR DESCRIPTION
**Pull request by LionelCons**
Follow-up from PR#147...

### CODE REVIEW

File: manifests/broker.pp [hunk 1]
```diff
@@ -69,6 +72,7 @@
   $group_id                                  = $kafka::params::group_id,
   $user_id                                   = $kafka::params::user_id,
   $config_dir                                = $kafka::params::config_dir,
+  $bin_dir                                   = $kafka::params::bin_dir,
```
Comment (bastelfreak) [inline manifests/broker.pp]: @LionelCons can you use the datatype Stdlib::Absolutepath here and in all other locations where you use the new parameter?

**Comment by bastelfreak** [comment]
This fixes https://github.com/voxpupuli/puppet-kafka/issues/145

**Comment by bastelfreak** [comment]
This replaces https://github.com/voxpupuli/puppet-kafka/pull/147

File: manifests/broker.pp (hunk 1 above)
Comment (bastelfreak) [inline manifests/broker.pp]: of course you can change all of them. But I want to have datatypes at least for all new parameters.

File: manifests/broker/service.pp [hunk 2]
```diff
@@ -17,6 +17,7 @@
   $heap_opts                  = $kafka::broker::heap_opts,
   $opts                       = $kafka::broker::opts,
   $config_dir                 = $kafka::broker::config_dir,
+  $bin_dir                    = $kafka::broker::bin_dir,
```
Comment (bastelfreak) [inline manifests/broker/service.pp]: we need "Stdlib::Absolutepath " here as well

File: manifests/mirror/service.pp [hunk 3]
```diff
@@ -18,6 +18,7 @@
   $blacklist                  = $kafka::mirror::blacklist,
   $max_heap                   = $kafka::mirror::max_heap,
   $config_dir                 = $kafka::params::config_dir,
+  $bin_dir                    = $kafka::params::bin_dir,
```
Comment (bastelfreak) [inline manifests/mirror/service.pp]: Stdlib::Absolutepath

File: manifests/consumer/service.pp [hunk 4]
```diff
@@ -12,6 +12,7 @@
   $service_defaults           = $kafka::consumer::service_defaults,
   $service_requires_zookeeper = $kafka::consumer::service_requires_zookeeper,
   $limit_nofile               = $kafka::consumer::limit_nofile,
+  $bin_dir                    = $kafka::consumer::bin_dir,
```
Comment (bastelfreak) [inline manifests/consumer/service.pp]: we need "Stdlib::Absolutepath " here as well

File: manifests/params.pp [hunk 5]
```diff
@@ -18,6 +18,7 @@
   $scala_version  = '2.11'
   $install_dir    = "/opt/kafka-${scala_version}-${version}"
   $config_dir     = '/opt/kafka/config'
+  $bin_dir        = '/opt/kafka/bin'
```
Comment (bastelfreak) [inline manifests/params.pp]: we need "Stdlib::Absolutepath " here as well

File: manifests/producer/service.pp [hunk 6]
```diff
@@ -12,6 +12,7 @@
   $service_config             = $kafka::producer::service_config,
   $service_defaults           = $kafka::producer::service_defaults,
   $service_requires_zookeeper = $kafka::producer::service_requires_zookeeper,
+  $bin_dir                    = $kafka::producer::bin_dir,
```
Comment (bastelfreak) [inline manifests/producer/service.pp]: we need "Stdlib::Absolutepath " here as well

**Comment by LionelCons** [comment]
@bastelfreak I've done for `$bin_dir` what you have done for the other parameters: define the type only for the public classes and not for the private/internal ones.

Compare for instance `broker.pp`:
```
  Boolean $service_install                   = $kafka::params::broker_service_install,
  Enum['running', 'stopped'] $service_ensure = $kafka::params::broker_service_ensure,
```
with `broker/service.pp`:
```
  $service_install            = $kafka::broker::service_install,
  $service_ensure             = $kafka::broker::service_ensure,
```

**Comment by bastelfreak** [comment]
Hi @LionelCons,
we had a discussion on IRC and decided that it is a better approach to add the types to all classes, even the private ones. This ensures consistency. Can you please add the Stdlib::Absolutepath to all occurences of $bin_dir?

**Comment by LionelCons** [comment]
@bastelfreak I've added the type even for private classes. 

IMHO, there are solutions to simplify the classes and parameters but let's discuss this elsewhere.

Please accept this PR so that we can move forward.

File: manifests/params.pp (hunk 5 above)
Comment (bastelfreak) [inline manifests/params.pp]: ah ups, we don't. all good

**Comment by bastelfreak** [comment]
@LionelCons let me know if you need this in a release, we can kick it of pretty quickly.

**Comment by LionelCons** [comment]
@bastelfreak In fact, there are several things that we would need to get fixed/improved. So no need for a new release now...

**Pull request merged**